src/org/webonise/multithreading/waitandnotify/Application.java

```diff
@@ -0,0 +1,22 @@
+package org.webonise.multithreading.waitandnotify;
+
+public class Application {
+
+    public void start(){
+
+        Thread thread = new Thread(new WorkerThread());
+        System.out.println("In main thread. Launching worker thread ...");
+        thread.start();
+
+        synchronized (thread){
```

Comment:
Am  bit worried here to pass `thread` object to synchronised block, what if thread's execution is complete before its lock is acquired or whenever we are trying to acquire its lock?